### PR TITLE
Feature/add component content list

### PR DESCRIPTION
**Pull request by acassiovilasboas**
## Description

- add ContentList component
- Change views to ContentList in SettingsListItem

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## ContentList
Simulator Screenshot - iPhone 15 - 2024-09-04 at 17 55 29

## Change on SettingsListItem
Simulator Screenshot - iPhone 15 - 2024-09-04 at 17 54 16

## Checklist:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have made corresponding changes to the documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All new and existing tests passed

### CODE REVIEW

File: OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift [hunk 1]
```diff
@@ -180,8 +140,16 @@ extension OceanSwiftUI {
...
                         }
                     } else {
-                        leadingView
-                            .layoutPriority(1)
+                        ContentList { view in
+                            view.parameters.title = parameters.title
+                            view.parameters.description = parameters.description
+                            view.parameters.newDescription = parameters.newDescription
+                            view.parameters.newDescription = parameters.newDescription
+                            view.parameters.caption = parameters.caption
```
Comment (vromeiro) [inline OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift:148]: faltou o type nao?

**Review by vromeiro — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift [hunk 2]
```diff
@@ -1,6 +1,6 @@
 //
 //  OceanSwiftUI+SettingListItem.swift
-//  DGCharts
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift:3]: kkkkkkk

File: OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift [hunk 3]
```diff
@@ -180,8 +140,16 @@ extension OceanSwiftUI {
                             view.parameters.lines = 2
                         }
                     } else {
-                        leadingView
-                            .layoutPriority(1)
+                        ContentList { view in
+                            view.parameters.title = parameters.title
+                            view.parameters.description = parameters.description
+                            view.parameters.newDescription = parameters.newDescription
+                            view.parameters.newDescription = parameters.newDescription
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift:147]: newDescription tá duplicado na linha 146 e 147

**Review by renanmassaroto — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift (hunk 1 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift:148]: Faltou sim, tinha faltado tbm o descriptionColor.. ja ajustei tbm

File: OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift (hunk 3 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift:147]: Opaaa, valeuu.. Ficou sim, esse ai era para ser o type, ajustei.

File: OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift (hunk 2 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Components SwiftUI/SettingListItem/OceanSwiftUI+SettingListItem.swift:3]: XCode é fogo!

**Pull request merged**